src/server/game/Entities/Object/Object.cpp

```diff
@@ -1147,7 +1147,12 @@ bool WorldObject::IsWithinLOSInMap(const WorldObject* obj) const
         return false;
 
     float ox, oy, oz;
-    obj->GetPosition(ox, oy, oz);
+    float angle = obj->GetAngle(GetPositionX(), GetPositionY());
+    float range = std::min(GetObjectSize(), GetExactDist(obj->GetPositionX(), obj->GetPositionY(), obj->GetPositionZ()));
```

Comment:
If I'm understanding what you're doing right, IsWithinLosInMap with object is checking the position of the object and getting the edge based on the angle of the source. Then passing that into the co-ordinate version of the same function which will do the same for the source object. Like this:

![image](https://cloud.githubusercontent.com/assets/173349/10907078/f775eea8-821e-11e5-978f-5dc682809135.png)

In that case, should the above not be obj->GetObjectSize() ?

It's possible I'm misunderstanding things a bit. But if the picture is right, maybe the above should be changed.

Overall this is a good thing. I've seen the LoS issues for a while and it was on my list of things to "look at" but admittedly quite some way down :)